doc/source/conf.py

```diff
@@ -128,3 +138,76 @@
     "body": generate_404(),
 }
 notfound_no_urls_prefix = True
+
+
+# specify the URL of the archive here
+# Url for sphinx-design
+archive_url = "https://github.com/executablebooks/sphinx-design/tree/main/docs/snippets/rst"
+
+
+def extract_example_links(archive_url: str) -> List[str]:
+    """
+    Extract example links from a specific URL.
+
+    Parameters
+    ----------
+    archive_url : str
+        The URL of the archive to retrieve the example links from.
+
+    Returns
+    -------
+    list
+        List of example links.
+    """
+    response = requests.get(archive_url)
+    soup = BeautifulSoup(response.content, "html5lib")
+    links = soup.find_all("a")
+    example_links = [
+        "https://raw.githubusercontent.com" + link["href"].replace("/blob/", "/")
+        for link in links
+        if link["href"].endswith(".txt") and not "article-info.txt" in link["href"]
+    ]
```

Comment:
Very good usage of list comprehensions. Let me suggests to add a new function parameter named `exclude_files`. It should be a list of strings. Filenames declared in this list should be excluded in line `168`.